packages/react/runtime/__test__/typecheck/jsx-runtime.test-d.tsx

```diff
@@ -0,0 +1,100 @@
+// Copyright 2024 The Lynx Authors. All rights reserved.
+// Licensed under the Apache License Version 2.0 that can be found in the
+// LICENSE file in the root directory of this source tree.
+import type { JSX } from '../../jsx-runtime/index.js';
+import type { MainThread, NodesRef, TouchEvent } from '@lynx-js/types';
+import { assertType, describe, test } from 'vitest';
+import { useMainThreadRef } from '../../lib/lynx-api.js';
+import { useRef } from '../../lib/hooks/react.js';
+
+describe('JSX Runtime Types', () => {
+  test('should support basic JSX element', () => {
+    const viewEle = (
+      <view>
+        <text>child node</text>
+      </view>
+    );
+    assertType<JSX.Element>(viewEle);
+  });
+
+  test('should validate the required props for raw-text', () => {
+    // @ts-expect-error: Missing required prop 'text'
+    const shouldError = <raw-text></raw-text>;
+
+    const rawTextELe = <raw-text text={'text'}></raw-text>;
+    assertType<JSX.Element>(rawTextELe);
+  });
+
+  test('should support JSX.Elements', () => {
+    function App() {
+      function renderFoo(): JSX.Element {
+        return <text></text>;
+      }
+      return renderFoo();
+    }
+    assertType<JSX.Element>(App());
+  });
+
+  test('should error on unsupported tags', () => {
+    // @ts-expect-error: Unsupported tag
+    const divElement = <div></div>;
+  });
+
+  test('should support event handlers', () => {
+    const viewWithBind = (
+      <view
+        bindtap={(e: TouchEvent) => {
```

Comment:
Can we use `assertType<number>(e.detail.x)` to test about event type instead of `TouchEvent`?